datalad/interface/run.py

```diff
@@ -443,7 +474,7 @@ def run_command(cmd, dataset=None, inputs=None, outputs=None, expand=None,
 """
     msg = msg.format(
         message if message is not None else _format_cmd_shorty(cmd),
-        json.dumps(run_info, indent=1, sort_keys=True, ensure_ascii=False))
+        '"{}"'.format(record_id) if sidecar else record)
```

Comment:
What about instead of "hard-coding" for a single possible sidecar file, we right away extend with the support for (possibly) multiple by extending current run record with additional field (e.g.`"sidecars"`) which would be a dict ({"backend": "filename"}) or a list (just["filenames"])?  Then those sidecar'ed files would be loaded to update current record, which could be entirely empty (besides `"sidecars"` field) or even partially filled in?

This might allow in the future, when we might decide on supporting various "runners" for them to "link" their "sidecar" report files to the run.  E.g. (hypothesizing)  `timeit` runner to the time records on how long it took, `niceman` could point to the environment description file, some `pbs-runner` to stdout/stderr files for the job(s), ...

Reply:
Sounds good. But it would have to be a dict under 'sidecars' as `rerun` needs to be able to load this info again and has to put it somewhere in the structure where we know how to find it again.

Thinking this further, we should probably take out the `diff` reporting in `rerun` and move it to such a complementary runner, such that a user could decide to store a full recursive diff alongside the initial run record in a sidecar file.

I like that.

Reply:
Gave it one more thought and changed my mind again: I prefer to keep it the way it is (only the hash in the commit message). The functionality you proposed can be achieved using an orthogonal approach. The run record can always have a 'sidecars' field in the way you described it &mdash; whether or not it is stored in the commit message. Stuff that works with the run records (like `rerun --report`) can then pull out the info from the additional sidecar files, **after** the core record is loaded from either message or file.

Otherwise we would always load a record from a message then come up with conditions when to fully or partially replace this record from one of the sidecar files. I think the "all or nothing (and everything else later)" approach is cleaner and leaner to implement without imposing limitations on further extensions.